libs/ui/src/modals/screens/tokens-selector/tokens-selector.tsx

```diff
@@ -72,7 +79,11 @@ export const TokensSelector: FC = () => {
     const { price } = allTokensMarketInfoSelector[tokenMetadataSlug] ?? {};
     const amountInDollar = getDollarValue({ amount, price, decimals });
 
-    const onSelectItem = () => navigate(ScreensEnum.SendToken, { token: item });
+    const onSelectItem = () => {
+      if (previousScreen === ScreensEnum.Swap || previousScreen === ScreensEnum.SendToken) {
+        navigate(previousScreen, { [field]: item });
+      }
+    };
```

Comment:
why you don't just `goBack` on each select?

Reply:
because on Swap we have two Token Selector and that's why I need to distinguish them somehow

Reply:
`goBack` will navigate back by your history, how it breaks your flow?

Reply:
![image](https://user-images.githubusercontent.com/35599200/214314618-18010286-d10c-4788-846f-ff964e52cb79.png)
<img width="866" alt="image" src="https://user-images.githubusercontent.com/35599200/214314786-34ecfadf-f6af-4089-9254-58a81e38bc40.png">
if I go to save Token for `fromToken` and `toToken` ?